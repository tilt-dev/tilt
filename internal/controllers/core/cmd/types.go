package cmd

import "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"

type Cmd = v1alpha1.Cmd
type CmdSpec = v1alpha1.CmdSpec
type CmdStatus = v1alpha1.CmdStatus
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdStateTerminated = v1alpha1.CmdStateTerminated

const LabelManifest = v1alpha1.LabelManifest
