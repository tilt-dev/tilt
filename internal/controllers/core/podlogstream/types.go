package podlogstream

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type PodLogStreamStatus = v1alpha1.PodLogStreamStatus
type ContainerLogStreamStatus = v1alpha1.ContainerLogStreamStatus
type ObjectMeta = metav1.ObjectMeta
