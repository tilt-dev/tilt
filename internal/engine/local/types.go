package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type ObjectMeta = metav1.ObjectMeta
