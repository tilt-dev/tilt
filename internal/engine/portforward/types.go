package portforward

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PortForward = v1alpha1.PortForward
type PortForwardSpec = v1alpha1.PortForwardSpec
type PortForwardStatus = v1alpha1.PortForwardStatus
type ObjectMeta = metav1.ObjectMeta
type Forward = v1alpha1.Forward
