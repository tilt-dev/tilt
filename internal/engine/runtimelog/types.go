package runtimelog

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type ObjectMeta = metav1.ObjectMeta
