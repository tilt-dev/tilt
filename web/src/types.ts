import { Alert } from "./alerts"
import { Facet } from "./facets"

export enum SocketState {
  Loading,
  Reconnecting,
  Closed,
  Active,
}

export enum ResourceView {
  Log,
  Alerts,
  Facets = 2,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManualAfterInitial,
  TriggerModeManualIncludingInitial,
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
  NotApplicable = "not_applicable",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  Building, // Tilt is actively doing something (e.g., docker build or kubectl apply)
  Pending, // not building, healthy, or unhealthy, but presumably on its way to one of those (e.g., queued to build, or ContainerCreating)
  Healthy, // e.g., build succeeded and pod is running and healthy
  Unhealthy, // e.g., last build failed, or CrashLoopBackOff
  Warning, // e.g., an undismissed restart
  None, // e.g., a manual build that has never executed
}

export type SnapshotHighlight = {
  beginningLogID: string
  endingLogID: string
  text: string
}

export enum ShowFatalErrorModal {
  Default,
  Show,
  Hide,
}

export type Snapshot = {
  view: Proto.webviewView
  isSidebarClosed: boolean
  path?: string
  snapshotHighlight?: SnapshotHighlight | null
}

// A plaintext representation of a line of the log,
// with metadata to render it in isolation.
//
// The metadata should be stored as primitive fields
// so that React's default caching behavior will kick in.
export type LogLine = {
  // We assume that 'text' does not contain a newline
  text: string
  manifestName: string
  level: string
}
