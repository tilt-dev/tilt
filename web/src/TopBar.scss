@import "constants";

.TopBar {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  padding-right: $sidebar-width;
  background-color: $color-gray-darkest;
  box-shadow: inset 0px -2px 10px 0px rgba($color-black, $translucent);
  z-index: $z-topbar;
  color: $color-gray-light;

  display: flex;
  flex-direction: row;
  align-items: center;
}

// Collapse/Expand
.TopBar-spacer {
  flex-grow: 1;
  height: 0;
}
