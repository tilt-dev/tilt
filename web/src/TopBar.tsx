import React, { PureComponent } from "react"
import { ResourceView } from "./types"
import "./TopBar.scss"
import SailInfo from "./SailInfo"
import TabNav from "./TabNav"

type TopBarProps = {
  previewUrl: string
  logUrl: string
  errorsUrl: string
  resourceView: ResourceView
  sailEnabled: boolean
  sailUrl: string
}

class TopBar extends PureComponent<TopBarProps> {
  render() {
    return (
      <div className="TopBar">
        <TabNav
          previewUrl={this.props.previewUrl}
          logUrl={this.props.logUrl}
          errorsUrl={this.props.errorsUrl}
          resourceView={this.props.resourceView}
        />
        <span className="TopBar-spacer">&nbsp;</span>
        <SailInfo
          sailEnabled={this.props.sailEnabled}
          sailUrl={this.props.sailUrl}
        />
      </div>
    )
  }
}

export default TopBar
