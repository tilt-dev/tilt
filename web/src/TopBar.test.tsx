import React from "react"
import renderer from "react-test-renderer"
import { MemoryRouter } from "react-router"
import { ResourceView } from "./types"
import TopBar from "./TopBar"

it("shows sail share button", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TopBar
          logUrl="/r/foo"
          previewUrl="/r/foo/preview"
          errorsUrl="/r/foo/errors"
          resourceView={ResourceView.Errors}
          sailEnabled={true}
          sailUrl=""
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it("shows sail url", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TopBar
          logUrl="/r/foo"
          previewUrl="/r/foo/preview"
          errorsUrl="/r/foo/errors"
          resourceView={ResourceView.Errors}
          sailEnabled={true}
          sailUrl="www.sail.dev/xyz"
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
