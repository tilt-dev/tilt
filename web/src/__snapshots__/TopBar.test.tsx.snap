// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows sail share button 1`] = `
<div
  className="TopBar"
>
  <nav
    className="TabNav"
  >
    <ul>
      <li>
        <a
          className="tabLink "
          href="/r/foo"
          onClick={[Function]}
        >
          Logs
        </a>
      </li>
      <li>
        <a
          className="tabLink "
          href="/r/foo/preview"
          onClick={[Function]}
        >
          Preview
        </a>
      </li>
      <li>
        <a
          className="tabLink tabLink--is-selected"
          href="/r/foo/errors"
          onClick={[Function]}
        >
          Errors
        </a>
      </li>
    </ul>
  </nav>
  <span
    className="TopBar-spacer"
  >
     
  </span>
  <span
    className="SailInfo"
  >
    <button
      type="button"
    >
      Share me!
    </button>
  </span>
</div>
`;

exports[`shows sail url 1`] = `
<div
  className="TopBar"
>
  <nav
    className="TabNav"
  >
    <ul>
      <li>
        <a
          className="tabLink "
          href="/r/foo"
          onClick={[Function]}
        >
          Logs
        </a>
      </li>
      <li>
        <a
          className="tabLink "
          href="/r/foo/preview"
          onClick={[Function]}
        >
          Preview
        </a>
      </li>
      <li>
        <a
          className="tabLink tabLink--is-selected"
          href="/r/foo/errors"
          onClick={[Function]}
        >
          Errors
        </a>
      </li>
    </ul>
  </nav>
  <span
    className="TopBar-spacer"
  >
     
  </span>
  <span
    className="SailInfo"
  >
    <a
      href="www.sail.dev/xyz"
    >
      Share this view!
    </a>
  </span>
</div>
`;
